property.cpp

```diff
@@ -615,9 +620,471 @@ namespace RayGene3D
+          //return (hash0 ^ (hash1 << 1));
+        };
+        auto compare_vertex_fn = [](const Vertex& l, const Vertex& r)
+        {
+          return (memcmp(&l, &r, sizeof(Vertex)) == 0);
+        };
+        std::unordered_map<Vertex, uint32_t, decltype(hash_vertex_fn), decltype(compare_vertex_fn)> vertex_map(8, hash_vertex_fn, compare_vertex_fn);
+
+        const auto& remap_vertex_fn = [&vertex_map](std::vector<Vertex>& vertices, const Vertex& vertex)
+        {
+          const auto result = vertex_map.emplace(vertex, uint32_t(vertices.size()));
+          if (result.second)
+          {
+            vertices.push_back({ vertex.pos, vertex.u, vertex.nrm, vertex.v });
+          }
+          return result.first->second;
+        };
+
+        const auto& gltf_indices = gltf_model.accessors[gltf_primitive.indices];
+        BLAST_ASSERT(gltf_indices.type == TINYGLTF_TYPE_SCALAR);
+
+        const auto& indices_view = gltf_model.bufferViews[gltf_indices.bufferView];
+        const auto indices_length = indices_view.byteLength;
+        const auto indices_offset = indices_view.byteOffset;
+        const auto indices_stride = indices_view.byteStride;
+
+        const auto& indices_buffer = gltf_model.buffers[indices_view.buffer];
+        
+
+        if (gltf_indices.componentType == TINYGLTF_COMPONENT_TYPE_INT || gltf_indices.componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_INT)
+        {
+          const auto indices_data = reinterpret_cast<const uint32_t*>(&indices_buffer.data[gltf_indices.byteOffset]);
+
+          for (uint32_t k = 0; k < gltf_indices.count / 3; ++k)
+          {
+            const auto vtx0 = create_vertex_fn(indices_data[k * 3 + 0], position_scale, 
+              pos_data, pos_stride, nrm_data, nrm_stride, tc0_data, tc0_stride);
+            const auto vtx1 = create_vertex_fn(indices_data[k * 3 + 1], position_scale,
+              pos_data, pos_stride, nrm_data, nrm_stride, tc0_data, tc0_stride);
+            const auto vtx2 = create_vertex_fn(indices_data[k * 3 + 2], position_scale,
+              pos_data, pos_stride, nrm_data, nrm_stride, tc0_data, tc0_stride);
+
+            const auto dp_10 = vtx1.pos - vtx0.pos;
+            const auto dp_20 = vtx2.pos - vtx0.pos;
+            const auto ng = glm::cross(dp_10, dp_20);
+            if (glm::length(ng) == 0.0f)
+            {
+              ++degenerated_geom_prim_count;
+              continue;
+            }
+
+            auto has_textures = false;
+            //has_textures = obj_materials[obj_mesh.material_ids[j]].diffuse_texname.empty() ? has_textures : true;
+            //has_textures = obj_materials[obj_mesh.material_ids[j]].alpha_texname.empty() ? has_textures : true;
+            //has_textures = obj_materials[obj_mesh.material_ids[j]].specular_texname.empty() ? has_textures : true;
+            //has_textures = obj_materials[obj_mesh.material_ids[j]].bump_texname.empty() ? has_textures : true;
+
+            if (has_textures)
```

Comment:
currently this condition is always false and we don't check the degenerate uv-wraps.